eng/common/scripts/Detect-Api-Changes.ps1

```diff
@@ -103,32 +102,38 @@ if (!($FindArtifactForApiReviewFn -and (Test-Path "Function:$FindArtifactForApiR
 }
 
 $responses = @{}
-foreach ($artifact in $ArtifactList)
+
+$packageProperties = Get-ChildItem -Recurse "$configFileDir" *.json
+
+foreach ($packagePropFile in $packageProperties)
 {
-    Write-Host "Processing $($artifact.name)"
-    $packages = &$FindArtifactForApiReviewFn $ArtifactPath $artifact.name
+    $packageMetadata = Get-Content $packagePropFile | ConvertFrom-Json
+    Write-Host "Processing $($packageMetadata.Name)"
+
+    $packages = &$FindArtifactForApiReviewFn $ArtifactPath $packageMetadata.Name
+
     if ($packages)
     {
         $pkgPath = $packages.Values[0]
-        $isRequired = Should-Process-Package -pkgPath $pkgPath -packageName $artifact.name
-        Write-Host "Is API change detect required for $($artifact.name):$($isRequired)"
+        $isRequired = Should-Process-Package -pkgPath $pkgPath -packageName $($packageMetadata.Name)
```

Comment:
I think this will be an issue for JS. For e.g. artifact name for JS package `@azure/core-auth` is `azure-core-auth`. Based on the modified logic it will look for a file with `@azure/core-auth.api.json` and it will fail with invalid file. Can we use `$($packageMetadata.ArtifactName)` instead of `$($packageMetadata.Name)` in places where it was referring `$artifact.name`

Reply:
I think I can. I'll debug and check out what goes wrong.

Reply:
[Build Run](https://dev.azure.com/azure-sdk/public/_build?definitionId=7050)

Commit the asked for updates 👍